dev/breeze/src/airflow_breeze/commands/release_candidate_command.py

```diff
@@ -32,7 +32,179 @@
+    console_print(f"[info]Validating version branches exist for {version_branch}...")
+
+    # Check if test branch exists
+    test_branch = f"v{version_branch}-test"
+    stable_branch = f"v{version_branch}-stable"
+
+    # Fetch to get latest remote branches
+    run_command(["git", "fetch", remote_name], check=True)
+
+    # Check remote branches
+    result = run_command(
+        ["git", "branch", "-r"],
+        check=True,
+        capture_output=True,
+        text=True,
+        dry_run_override=False,
+    )
+    remote_branches = result.stdout
+
+    test_branch_exists = f"{remote_name}/{test_branch}" in remote_branches
+    stable_branch_exists = f"{remote_name}/{stable_branch}" in remote_branches
+
+    if not test_branch_exists:
+        console_print(f"[error]Test branch '{remote_name}/{test_branch}' does not exist!")
+        console_print("Available remote branches:")
+        run_command(["git", "branch", "-r"])
+        exit(1)
+    else:
+        console_print(f"[success]Test branch '{remote_name}/{test_branch}' exists")
+
+    if not stable_branch_exists:
+        console_print(f"[error]Stable branch '{remote_name}/{stable_branch}' does not exist!")
+        console_print("Available remote branches:")
+        run_command(["git", "branch", "-r"])
+        exit(1)
+    else:
+        console_print(f"[success]Stable branch '{remote_name}/{stable_branch}' exists")
+
+
+def validate_tag_does_not_exist(version, remote_name):
+    """Validate that the release tag doesn't already exist."""
+    console_print(f"[info]Checking if tag '{version}' already exists...")
+
+    # Check if tag exists locally
+    local_result = run_command(
+        ["git", "tag", "-l", version],
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+
+    # Check if tag exists on remote using ls-remote with --exit-code
+    remote_result = run_command(
+        ["git", "ls-remote", "--exit-code", "--tags", remote_name, f"refs/tags/{version}"],
+        check=False,
+        capture_output=True,
+        text=True,
+    )
+
+    tag_exists_locally = bool(local_result.stdout.strip())
+    tag_exists_remotely = remote_result.returncode == 0
+
+    if tag_exists_locally or tag_exists_remotely:
+        location = []
+        if tag_exists_locally:
+            location.append("locally")
+        if tag_exists_remotely:
+            location.append("remotely")
+
+        console_print(f"[error]Tag '{version}' already exists {' and '.join(location)}!")
+
+        if tag_exists_locally:
+            console_print(f"Use 'git tag -d {version}' to delete it locally if needed")
+        if tag_exists_remotely:
+            console_print(f"Use 'git push {remote_name} --delete {version}' to delete it remotely if needed")
+
+        if not confirm_action("Do you want to continue anyway? This may cause issues."):
+            exit(1)
+    else:
+        console_print(f"[success]Tag '{version}' does not exist yet")
```

Comment:
```suggestion
    if not tag_exists_locally and not tag_exists_remotely:
        console_print(f"[success]Tag '{version}' does not exist yet")
        return

    location = []
    if tag_exists_locally:
        location.append("locally")
    if tag_exists_remotely:
        location.append("remotely")

    console_print(f"[error]Tag '{version}' already exists {' and '.join(location)}!")

    if tag_exists_locally:
        console_print(f"Use 'git tag -d {version}' to delete it locally if needed")
    if tag_exists_remotely:
        console_print(f"Use 'git push {remote_name} --delete {version}' to delete it remotely if needed")

    if not confirm_action("Do you want to continue anyway? This may cause issues."):
        exit(1)
```

nit, invert it then you can short circuit out early and reduce nesting

Reply:
Fixed in https://github.com/apache/airflow/pull/54960/commits/769435bffc94d9b857c3c3b008dade6a43b01f1e